network/src/peers/peers.rs

```diff
@@ -48,27 +47,31 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
         // give us 100ms to close some negatively judge_badd connections (probably less needed, but we have time)
         tokio::time::sleep(Duration::from_millis(100)).await;
 
-        // Attempt to connect to the default bootnodes of the network.
-        self.connect_to_bootnodes().await;
-
-        // Attempt to connect to each disconnected peer saved in the peer book.
-        if !self.config.is_bootnode() {
-            self.connect_to_disconnected_peers().await;
-        }
-
-        // Broadcast a `GetPeers` message to request for more peers.
-        self.broadcast_getpeers_requests().await;
-
-        let new_active_peer_count = self.peer_book.get_active_peer_count() as usize;
-        // Check if this node server is above the permitted number of connected peers.
-        let max_peers = self.config.maximum_number_of_connected_peers() as usize;
-        if new_active_peer_count > max_peers {
-            let number_to_disconnect = new_active_peer_count - max_peers;
-            trace!(
-                "Disconnecting from {} peers to maintain their permitted number",
-                number_to_disconnect
-            );
+        // Fetch active peer count after high RTTs disconnects.
+        let active_peer_count = self.peer_book.get_active_peer_count();
+        let min_peers = self.config.minimum_number_of_connected_peers() as u32;
+        let max_peers = self.config.maximum_number_of_connected_peers() as u32;
+
+        // Calculate the peer counts to disconnect and connect based on the node type and current
+        // peer counts.
+        let (number_to_disconnect, number_to_connect) = match self.config.is_bootnode() {
+            true => (
+                // Bootnodes disconnect down to the min peer count, this to free up room for
+                // the next crawled peers...
+                active_peer_count.saturating_sub(min_peers),
+                // ...then they connect to peers up to min peers below the max, so as to keep room for
+                // new incoming connections.
+                max_peers.saturating_sub(2 * min_peers),
```

Comment:
this is twice the number specified in the comment; also, shouldn't it depend on `active_peer_count` too?

Reply:
I've modified this in [77250a0](https://github.com/AleoHQ/snarkOS/pull/878/commits/77250a0d88231ce4bdcb67dac65b25f7a3067be2) to use fill up to 80% of the node's capacity for crawling.